examples/parallel-optimization.ipynb

```diff
@@ -0,0 +1,205 @@
+    "from skopt import Optimizer\n",
+    "\n",
+    "# just a normal optimizer instantiation \n",
+    "optimizer = Optimizer(\n",
+    "    base_estimator=ExtraTreesRegressor(),\n",
+    "    dimensions=[Real(-3.0, 3.0) for i in range(3)],\n",
+    "    acq_optimizer='sampling'\n",
+    ")\n",
+    "\n",
+    "x = optimizer.ask(n_points=2, parallel_strategy='cl_min')\n",
+    "\n",
+    "print(x)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "With `n_points=2`, the result of ask method is a list of two points for you to evaluate. You can use for example the joblib package to evaluate these points in parallel like so:\n",
+    "\n",
+    "`Parallel()(delayed(objective)(v) for v in x)`\n",
+    "\n",
+    "You can also specify different strategies that skopt can use to generate n points to evaluate in parallel. Currently the \"minimum constant liar\" strategy is used; You can see how it works by following the link at the bottom of this notebook. You can see the list of supported strategies in the definition of the `ask` method. \n",
+    "\n",
+    "FYI: points in the output of ask will always be different, and if they are not, please report this as an issue. Notice also that if you set `n_points=1`, then ask will also return a list, but with a single point to evaluate.\n",
+    "\n",
+    "Lets compare a sequential mode of optimization with `n_points=1` to a parallel mode. For this, lets write a function that plots results of optimization at every sequential step:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 34,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# objective function to minimze\n",
+    "def objective(x):\n",
+    "    return np.sum(np.array(x) ** 2)\n",
+    "\n",
+    "def plot_optimization(n_jobs):\n",
+    "    optimizer = Optimizer(\n",
+    "        base_estimator=ExtraTreesRegressor(),\n",
+    "        dimensions=[Real(-3.0, 3.0) for i in range(3)],\n",
+    "        acq_optimizer='sampling'\n",
+    "    )\n",
+    "\n",
+    "    # configure number of workers to be used in parallel, and overall # of computations\n",
+    "    n_steps, Y = 20, []\n",
+    "\n",
+    "    for i in range(n_steps):\n",
+    "        x = optimizer.ask(n_points=n_jobs)\n",
+    "        # evaluate n_points in parallel\n",
+    "        y = Parallel()(delayed(objective)(v) for v in x)\n",
+    "        # tell points and corresponding objectives to the optimizer\n",
+    "        optimizer.tell(x, y)\n",
+    "        # keep objectives history\n",
+    "        Y.append(min(optimizer.yi))\n",
+    "    \n",
+    "    # plot results of optimization \n",
+    "    plt.plot(range(len(Y)), Y, label=\"points=\" + str(n_jobs))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now lets use the function and see how sequential mode performs against the parallel mode:"
```

Comment:
Yeah, I do not like how I structured the example and the text as such. I will rewrite taking into account your suggestions.